### PR TITLE
Replace semantic widget/link definitions

### PR DESCRIPTION
**Pull request by WilcoFiers**
Part of #1593 was to add a single definition that makes it more explicit when a we mean the exact role, or any role that inherits. This adds a new definition "inheriting semantic role" which replaces "semantic widget" and "semantic link". I'll open a separate PR to change the definition of `semantic role` to allow phrases like "semantic button".

Need for Final Call: 1 week.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: pages/glossary/inheriting-semantic-role.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+title: Inheriting Semantic Role
+key: inheriting-semantic
+unambiguous: true
+objective: true
+input_aspects:
+  - Accessibility tree
+  - DOM tree
+---
+
+An element with a _inheriting semantic role_ of X is any element which has the [semantic role][] of X or a [semantic role][] that inherits from the X.
```
Comment (WilcoFiers) [inline pages/glossary/inheriting-semantic-role.md:11]: I'm considering excluding abstract roles from this.... What do people think?

File: _rules/table-header-cell-has-assigned-cells-d0f69e.md [hunk 2]
```diff
@@ -27,16 +27,16 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any HTML element with the [semantic role][] of [rowheader][] or [columnheader][] for which all of the following is true:
+This rule applies to any HTML element with is a [semantic][semantic role] [rowheader][] or [columnheader][] for which all of the following is true:
```
Comment (Jym77) [inline _rules/table-header-cell-has-assigned-cells-d0f69e.md:30]: I feel this requires a precision in the def of "Semantic Role" that we can say "an element is a semantic foo" to mean "has a semantic role of foo".

**Review by Jym77 — CHANGES_REQUESTED** [review]

**Review by tbostic32 — APPROVED** [review]

File: _rules/table-header-cell-has-assigned-cells-d0f69e.md (hunk 2 above)
Comment (kherr-a11y) [inline _rules/table-header-cell-has-assigned-cells-d0f69e.md:30]: I don't have edit privileges yet, but "is a" - please remove the "is"

File: _rules/table-header-cell-has-assigned-cells-d0f69e.md [hunk 3]
```diff
@@ -27,16 +27,16 @@ acknowledgments:
...
 - the element is [included in the accessibility tree][]; and
-- the element has at least one ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]; and
-- the element's closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][] is [included in the accessibility tree][].
+- the element has at least one ancestor in the [flat tree][] that is a [semantic][semantic role] [table][] or [grid][]; and
+- the element's closest ancestor in the [flat tree][] that is a [semantic][semantic role] [table][] or [grid][] is [included in the accessibility tree][].
 
 ## Expectation
 
-Each target element is [assigned][] to at least one element with a [semantic role][] of [cell][] or inheriting from [cell][].
+Each target element is [assigned][] to at least one element this is an [inheriting semantic][] [cell][].
```
Comment (kherr-a11y) [inline _rules/table-header-cell-has-assigned-cells-d0f69e.md:39]: "this is" should be "with"

**Review by kherr-a11y — COMMENTED** [review]
two small editorial changes suggested.  I do not have edit privileges.

File: _rules/table-header-cell-has-assigned-cells-d0f69e.md (hunk 3 above)
Comment (WilcoFiers) [inline _rules/table-header-cell-has-assigned-cells-d0f69e.md:39]:
```suggestion
Each target element is [assigned][] to at least one element with an [inheriting semantic][] [cell][].
```

File: _rules/table-header-cell-has-assigned-cells-d0f69e.md (hunk 2 above)
Comment (daniel-montalvo) [reply _rules/table-header-cell-has-assigned-cells-d0f69e.md:30]:
```suggestion
This rule applies to any HTML element with a [semantic][semantic role] [rowheader][] or [columnheader][] for which all of the following is true:
```

Is that what you mean, @kherr-a11yb

Comment (Jym77) [reply _rules/table-header-cell-has-assigned-cells-d0f69e.md:30]: Solved in https://github.com/act-rules/act-rules.github.io/pull/1671/files#diff-a0b8349df87c854df34f5481b6c6a983ca594f8b4ea62347239579f80d6d4e0fR16

**Review by Jym77 — APPROVED** [review]

**Review by daniel-montalvo — APPROVED** [review]

**Pull request merged**